hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SBaseStruct.scala

```diff
@@ -110,7 +110,7 @@ trait SBaseStructValue extends SValue {
   }
 
   def insert(cb: EmitCodeBuilder, region: Value[Region], newType: TStruct, fields: (String, EmitValue)*): SBaseStructValue = {
-    if (newType.size < 64 || fields.length < 16)
+    if (st.settableTupleTypes().length + fields.map(_._2.emitType.settableTupleTypes.length).sum < 64)
```

Comment:
Besides changing to measure struct size by code tuple size, this also changes from "new struct is small, or added fields are small" to just "new struct is small". Can you explain that change? Do we not want to use an InsertFieldsStruct when adding a few fields to a big struct? (Actually, why does the size of the original struct matter at all?)

Reply:
Makes sense, thanks!

Reply:
good question, should have clarified the reason in the initial comment!